### PR TITLE
Cumulus 1226 - Index from database

### PR DESCRIPTION
**Pull request by laurenfrederick**
**Summary:** Index to an elasticsearch index from the database

Addresses [CUMULUS-1226](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1226)

Documentation PR: https://github.com/nasa/cumulus-api/pull/266

## Changes

* Added `elasticsearch/index-from-database` endpoint to index from the database using async operations
* Added a lambda to perform a full index from the database
* Added `elasticsearch/indices-status` endpoint to see the status of the indices (just call the ES function)

## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [x] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/api/endpoints/elasticsearch.js [hunk 1]
```diff
@@ -177,10 +181,73 @@ async function changeIndex(req, res) {
...
+async function indicesStatus(req, res) {
+  const esClient = await Search.es();
+
+  return res.send(await esClient.cat.indices({}));
+}
+
+async function indexFromDatabase(req, res) {
+  const esClient = await Search.es();
+
+  const indexName = req.body.indexName || timestampedIndexName();
```
Comment (kkelly51) [inline packages/api/endpoints/elasticsearch.js]: What if there is no req.body.indexName and timestampedIndexName() is incorrect? Like if it was a different date?

File: packages/api/es/search.js [hunk 2]
```diff
@@ -22,7 +21,11 @@ const logDetails = {
 
 const defaultIndexAlias = 'cumulus-alias';
 
-const getCredentials = promisify(aws.config.getCredentials);
+const getCredentials = () =>
+  new Promise((resolve, reject) => aws.config.getCredentials((err) => {
```
Comment (kkelly51) [inline packages/api/es/search.js]: Why did we not like the promisify?

Comment (laurenfrederick) [reply packages/api/es/search.js]: Running in ECS it would not work

File: packages/api/endpoints/elasticsearch.js (hunk 1 above)
Comment (laurenfrederick) [reply packages/api/endpoints/elasticsearch.js]: I'm confused... can you give me an example?

File: packages/api/tests/lambdas/test-index-from-database.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  await rulesModel.deleteTable();
+
+  await aws.recursivelyDeleteS3Bucket(process.env.system_bucket);
+});
+
+test('No error is thrown if nothing is in the database', async (t) => {
+  t.notThrows(async () => indexFromDatabase.indexFromDatabase(esIndex, tables));
+});
+
+test.only('index executions', async (t) => {
```
Comment (kkelly51) [inline packages/api/tests/lambdas/test-index-from-database.js]: The `only` is still on. Did you mean this?

File: packages/api/tests/lambdas/test-index-from-database.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  const searchResults = await Promise.all([
+    searchEs('collection'),
+    searchEs('execution'),
+    searchEs('granule'),
+    searchEs('pdr'),
+    searchEs('provider'),
+    searchEs('rule')
+  ]);
+
+  searchResults.map((res) => t.is(res.meta.count, numItems));
```
Comment (kkelly51) [inline packages/api/tests/lambdas/test-index-from-database.js]: Good 👍

File: packages/api/endpoints/elasticsearch.js (hunk 1 above)
Comment (kkelly51) [reply packages/api/endpoints/elasticsearch.js]: Or okay, I guess this is creating the index so using the current date is fine but what about referencing the index later on? How would they get the correct indexName?

File: packages/api/tests/lambdas/test-index-from-database.js (hunk 3 above)
Comment (laurenfrederick) [reply packages/api/tests/lambdas/test-index-from-database.js]: definitely not!! this is why PR reviews are so great! thank you!

File: packages/api/endpoints/elasticsearch.js (hunk 1 above)
Comment (laurenfrederick) [reply packages/api/endpoints/elasticsearch.js]: there is the indices-status call where you can view all your indices. You used to be able to see it the in AWS console, but when we put ES in the VPC you can't see all your indices in the console which is why I created that API call that just pretty much calls ES. Then the `current-index` API call shows you the one you're currently using. I think you also used to be able to see that through the console.

Comment (kkelly51) [reply packages/api/endpoints/elasticsearch.js]: 👍 okay that's good

**Pull request merged**